lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -528,7 +528,11 @@ export abstract class ClientApplication {
         const atbcMeasurement = this.performanceClient.startMeasurement(PerformanceEvents.AcquireTokenByCode, request.correlationId);
 
         try {
-            if (request.code) {
+            if (request.code && request.nativeAccountId) {
+                // Throw error in case server returns both spa_code and spa_accountid in exchange for auth code.
+                throw AuthError.createUnexpectedError("Request cannot contain both SPA code and native account id.");
```

Comment:
Create a predefined error for this

Reply:
The user would not know how to handle it. We can create a pre-defined error but this would mean a STS bug which is by definition `unexpected`. I am good with creating a new one as long as we get this message across.

Reply:
That's fine. Any error we create should be predefined. Any error thrown by the Server should be thrown as is as `ServerError`

Reply:
Not sure I agree completely. `ClientConfigurationError` and `ClientAuthError` are msal generated errors. `Unexpected` is in the base class for cases like this. For this case, I guess we can make it a `ClientAuthError` with appropriate messaging. My issue is with the wrong message - It is not a `client` error but since we throw it, may be it is fine.

Reply:
I disagree. This _is_ a client error. The developer passed in both a code and an accountId to our API which expects one or the other. If the server returned both in their backend app it should be a server error on the backend.

Reply:
That is fair. Since the interface is with the developer, I am fine with this.

Reply:
Let us add docs on this later @lalimasharda.